### DISCUSSION TITLE
How to interpret gaps?

### DISCUSSION THREAD

**Discussion by cozos** [bloomberg/memray#265] [Q&A] [answered]
I have a memray flamegraph where a 2.5GB trace turns into 600MB (see screenshot). Where is the other 1.9GB going? Is it padding or something? How do I interpret this?

![Screen Shot 2022-12-06 at 11 55 04 PM](https://user-images.githubusercontent.com/2646862/206120441-0baa00af-7558-4a93-80fa-c7d52d3c7a85.png)

![Screen Shot 2022-12-06 at 11 39 29 PM](https://user-images.githubusercontent.com/2646862/206118993-3cdea3b8-f1a6-4ae0-a34f-fabeb7abd15a.png)

Flamegraph HTML file available here: https://gist.github.com/cozos/fb4579ced8a7dfdb51fdbf531ce3494f

Thanks

**Comment by pablogsal** [comment, marked as answer]
Thanks for checking with us. This aspect is explained in the docs:

> The width of each function box represents how much memory was allocated by that function or its children. Functions with wider boxes allocated more bytes of memory than those with narrower boxes, in proportion to their width.

This is also explained in one of the examples here:

> The top edge shows that function g() allocates the most memory, d() is wider, but its exposed top edge is smaller, which means that d() itself allocated less memory than the one allocated by the functions called by it. Functions including b() and c() do not allocate memory themselves directly; rather, the functions they called did the allocating.

that means that the function `process` and functions called by it allocates 2.5GB together. As there is only one caller and it only allocates 600MB it means that the function `process` allocates 1.9 GB by itself. Is very likely that these allocations are native allocations in C/C++ so you may want to try to use the `—native` option to get more information about where they are coming from.

Please check the docs https://bloomberg.github.io/memray/flamegraph.html#interpreting-flame-graphs for more information.

**Comment by cozos** [comment]
That helps a lot. Sorry I didn't read the very comprehensive docs! Will study it for sure!

Thanks!
